turbinia/lib/docker_manager.py

```diff
@@ -0,0 +1,222 @@
+    preconfigured Docker environment variables.
+
+    Returns:
+      docker_client(DockerClient): The Docker daemon client.
+
+    Raises:
+      TurbiniaException: If the Docker daemon could not be connected to.
+    """
+    try:
+      docker_client = docker.from_env()
+    except docker.errors.APIError as exception:
+      raise TurbiniaException(
+          'An issue has occured connecting to the Docker daemon: {0!s}'.format(
+              exception))
+    return docker_client
+
+  def verify_image(self, image_id):
+    """Verify that the Docker image exists.
+
+    Args:
+      image_id(str): The short image id to check for.
+
+    Returns:
+      image_verif(Image): The Image object.
+
+    Raises:
+      TurbiniaException: If the Docker image is not found.
+    """
+    try:
+      image_verif = self.client.images.get(image_id)
+    except docker.errors.ImageNotFound as exception:
+      message = 'The Docker image {0!s} could not be found: {1!s}'
+      raise TurbiniaException(message.format(image_id, exception))
+    return image_verif
+
+  def list_images(self, filters=None):
+    """Lists all available Docker images.
+
+    Args:
+     filters(str): If provided, will return a subset of the Images data.
+        Allowed values are 'short_id' and 'id'.
+
+    Returns:
+      images(list): A list of available Docker images.
+
+    Raises:
+      TurbiniaException: If an error occured retrieving the images.
+    """
+    accepted_vars = ['short_id', 'id']
+    try:
+      images = self.client.images.list()
+      if filters in accepted_vars:
+        # short_id and id will always start with sha256:
+        images = [
+            getattr(img, filters).replace('sha256:', '') for img in images
+        ]
+    except docker.errors.APIError as exception:
+      raise TurbiniaException(
+          'An error occured retrieving the images: {0!s}'.format(exception))
+    return images
+
+
+class ContainerManager(DockerManager):
+  """Class representing Docker containers.
+
+  Attributes:
+    image_id(str): Docker image id.
+  """
+
+  def __init__(self, image_id):
+    """Initialize the ContainerManager object.
+
+    Args:
+      image_id(str): The image id to create a container from.
+    """
+    super(ContainerManager, self).__init__()
+    self.image_id = self.verify_image(image_id)
+
+  def _create_mount_points(self, mount_paths):
+    """Creates file and device mounting arguments.
+
+    The arguments will be passed into the container and all device blocks
+    will be mounted as ro while file paths would be mounted as rw.
+
+    Attributes:
+      mount_paths(list): The paths on the host system to be mounted.
+
+    Returns:
+      device_paths(list): device blocks that will be mounted.
+      file_paths(dict): file paths that will be mounted.
+    """
+    device_paths = []
+    file_paths = {}
+
+    for mpath in mount_paths:
+      if mpath not in file_paths.keys() or mpath not in device_paths:
```

Comment:
Do you want an `and` here instead of an `or`? I assume this is to prevent duplicates?  I think a path will only end up in one or the other, so the body of the statement is always going to be executed.